_dprhtml/features/search/init.js

```diff
@@ -1,154 +1,168 @@
+    this.searchT = ko.observable(false)
+    this.searchBookString = ko.observable('')
+    this.searchRegex = ko.observable(false)
+
+    this.searchSetString = ko.observable('dmsak')
+    this.searchSetV = ko.observable(false)
+    this.searchSetD = ko.observable(true)
+    this.searchSetM = ko.observable(true)
+    this.searchSetS = ko.observable(true)
+    this.searchSetA = ko.observable(true)
+    this.searchSetK = ko.observable(true)
+    this.searchSetY = ko.observable(false)
+    this.searchSetX = ko.observable(false)
+    this.searchSetB = ko.observable(false)
+    this.searchSetG = ko.observable(false)
+    this.searchSetN = ko.observable(false)
+
+    this.bookMenu = ko.observableArray()
+    this.bookListA = ko.observableArray()
+    this.bookListB = ko.observableArray()
+
+    this.searchHierarchy = ko.observable([])
+
+    this.metaList = ko.observableArray()
+    this.volumeList = ko.observableArray()
+    this.vaggaList = ko.observableArray()
+    this.suttaList = ko.observableArray()
+    this.sectionList = ko.observableArray()
+
+    this.metaListValue = ko.observable('0')
+    this.volumeListValue = ko.observable('0')
+    this.vaggaListValue = ko.observable('0')
+    this.suttaListValue = ko.observable('0')
+    this.sectionListValue = ko.observable('0')
+    this.partialValue = ko.observable('1')
   }
 
-
-  searchPart(part){
-    let searchParts = part.toString().split('.');
-    this.partialValue(`${parseInt(searchParts[0])+1}`);
-    if (searchParts.length == 6){
-      this.metaListValue(searchParts[1]);
-      this.volumeListValue(searchParts[2]);
-      this.vaggaListValue(searchParts[3]);
-      this.suttaListValue(searchParts[4]);
-      this.sectionListValue(searchParts[5]);
+  searchPart(part) {
+    const searchParts = part.toString().split('.')
+    this.partialValue(`${parseInt(searchParts[0], 10) + 1}`)
+    if (searchParts.length === 6) {
+      this.metaListValue(searchParts[1])
+      this.volumeListValue(searchParts[2])
+      this.vaggaListValue(searchParts[3])
+      this.suttaListValue(searchParts[4])
+      this.sectionListValue(searchParts[5])
     }
   }
 
-  searchBookDropdown(){
-    if(this.searchBookString() == '' || DPR_G.searchType < 2){
-      return '1';
-    } else {
-      return this.searchBookString();
+  searchBookDropdown() {
+    if (this.searchBookString() === '' || DPR_G.searchType < 2) {
+      return '1'
     }
+
+    return this.searchBookString()
   }
 
-  searchBookCheckbox(bookNumber){
-    if(this.searchBookString() == '')
-    {
-      return true;
-    } else {
-      const bookArray = (this.searchBookString()||'').split(',');
-      return bookArray.includes(`${bookNumber}`);
+  searchBookCheckbox(bookNumber) {
+    if (this.searchBookString() === '') {
+      return true
     }
+
+    const bookArray = (this.searchBookString() || '').split(',')
+    return bookArray.includes(`${bookNumber}`)
   }
 
-  searchSet(set){
-    set = set.toLowerCase();
```

Comment:
I was just talking with someone the other day about why re-assigning parameters can be confusing. Thanks for cleaning this up!